Add release checklist (#551)
embedded-graphics/embedded-graphics: doc/releasing.md

# Release process

Target audience: crate maintainers who wish to release `embedded-graphics` or `embedded-graphics-core`.

> Please take a cautious approach to this. If any step doesn't feel right or doesn't succeed smoothly, stop and rectify any issues before continuing.

> ## On Github

Reviewer: Extremely important change :wink::
```suggestion
## On GitHub
```

Author: Absolutely vital! ;)

- Check that all desired PRs are merged and all desired issues are closed/resolved.
- Check that the latest master build passed in CircleCI.

## On your local machine

- `cd` to the repository root
- Use the crate MSRV of 1.40 by running `rustup override set 1.40`
- Check that `just` and `cargo-release` are installed and available in `$PATH`.
  - `just --version`
  - `cargo release --version`

> - Ensure you have the latest changes with `git checkout master` and `git pull`

Reviewer: Should we use the new `switch` command instead of `checkout`?
```suggestion
- Ensure you have the latest changes with `git switch master` and `git pull`
```

Author: TIL that's a command! My muscle memory has always used `checkout`.

> - Check that your local repository is clean (no uncommitted changes, no unpushed commits. Ideally, use `git reset --hard origin/master` to ensure your local state is up to date with `origin/master`.

Reviewer: This assumes that the `embedded-graphics/embedded-graphics` remote is called `origin`, which it might not if you set up the repo using the `gh` CLI tool. It should be clear what is meant, but perhaps we should add a note to replace `origin` by whatever the remote is actually called.

And the line is missing the closing `)`.

  - Add the crate name to the `Release title` field
  - Copy and paste the latest released section out of the crate's `CHANGELOG.md` file into the `Describe this release` field. Do not include the version header, e.g.:

    ```markdown
    ### Added

    - [#111](https://github.com/embedded-graphics/embedded-graphics/pull/111) Added something

    ### Removed

    - [#222](https://github.com/embedded-graphics/embedded-graphics/pull/222) Removed a thing
    ```

  - Checkbox For `alpha` or `beta` releases, check the `This is a pre-release` checkbox.
  - Hit <kbd>Publish release</kbd>
  - Check that the release is displayed on the [repository homepage](https://github.com/embedded-graphics/embedded-graphics).
  - Post a link to the released tag (e.g. <https://github.com/embedded-graphics/embedded-graphics/releases/tag/embedded-graphics-v0.7.0-alpha.3>) to the embedded-graphics Matrix room at <https://matrix.to/#/!SfJCDXZbMHXkPovtKL:matrix.org>
  - If you are @jamwaffles, post a Tweet tagging @rustembedded with a happy announcement message.

> - Check the other repositories in the [embedded-graphics organisation](https://github.com/embedded-graphics) for depenencies on `embedded-graphics` and `embedded-graphics-core`. The versions should be updated to the latest releases made whilst following this guide.

Reviewer: ```suggestion
- Check the other repositories in the [embedded-graphics organization](https://github.com/embedded-graphics) for dependencies on `embedded-graphics` and `embedded-graphics-core`. The versions should be updated to the latest releases made whilst following this guide.
```
